core/types/transaction.go

```diff
@@ -330,6 +334,34 @@ func (tx *Transaction) Cost() *big.Int {
 	return total
 }
 
+// RollupDataGas is the amount of gas it takes to confirm the tx on L1 as a rollup
+func (tx *Transaction) RollupDataGas() uint64 {
```

Comment:
This function will need to be hardforkable eventually

Reply:
Worst-case we deprecate the function, and call a different function. The l1 cost computation has access to both the chain config and current block number, so there it's easy to hardfork: https://github.com/ethereum-optimism/reference-optimistic-geth/blob/d9da8f7e57d9770fed3b79f537152985e28dc8cd/core/rollup_l1_cost.go#L49